File: solutions/cpp/vehicle-purchase/1/vehicle_purchase.cpp
#include "vehicle_purchase.h"

namespace vehicle_purchase {

    // needs_license determines whether a license is needed to drive a type of vehicle. Only "car" and "truck" require a license.
    bool needs_license(std::string kind){
		return kind.compare("truck") == 0 || kind.compare("car") == 0;
    }

    // choose_vehicle recommends a vehicle for selection. It always recommends the vehicle that comes first in lexicographical order.
    std::string choose_vehicle(std::string option1, std::string option2) {
		
		std::string choice = option1.compare(option2) <= 0 ? option1 : option2;
        return choice + " is clearly the better choice.";
    }

    // calculate_resell_price calculates how much a vehicle can resell for at a certain age.
    double calculate_resell_price(double original_price, double age) {
        double multiplier = age < 3 ?
			0.8
			: age < 10 ?
			0.7
			: 0.5;
			
        return original_price * multiplier;
    }

}  // namespace vehicle_purchase

File: solutions/go/vehicle-purchase/1/vehicle_purchase.go
package purchase

import "fmt"

// NeedsLicense determines whether a license is needed to drive a type of vehicle. Only "car" and "truck" require a license.
func NeedsLicense(kind string) bool {
	return kind == "car" || kind == "truck"
}

// ChooseVehicle recommends a vehicle for selection. It always recommends the vehicle that comes first in lexicographical order.
func ChooseVehicle(option1, option2 string) string {
	var vehicle string

	if option1 <= option2 {
		vehicle = option1
	} else {
		vehicle = option2
	}
	return fmt.Sprintf("%s is clearly the better choice.", vehicle)
}

// CalculateResellPrice calculates how much a vehicle can resell for at a certain age.
func CalculateResellPrice(originalPrice, age float64) float64 {
	var rate float64

	if age < 3.0 {
		rate = 0.8
	} else if age < 10.0 {
		rate = 0.7
	} else {
		rate = 0.5
	}
	return rate * originalPrice
}
